### PR TITLE
21 squad movement first pass

### PR DESCRIPTION
**Pull request by avelitj12**
Movement working as follows:

- Squads lock into 1 of 4 slots when entering dome
- Right clicking after select will only move squad if right click is in (fairly large) dome select hitbox and if dome is known by current dome to be connected
- When entering a dome, squad is assigned slot position to move to
- When exiting a dome, squad is teleported to dome center then given target to move to (sneaky way to keep them along the connectors visually)
- Selected squad is now highlighted in yellow

**Known issue**
- Some domes will activate a squad's "_on_area_entered" signal, causing the squad to be reassigned to an open slot and not leave the dome. Trying to move the squad again usually works the second time.

### CODE REVIEW

**Comment by avelitj12** [comment]
Latest commit resolves known issue

**Comment by cmavelis** [comment]
are these still true?
> - When exiting a dome, squad is teleported to dome center then given target to move to (sneaky way to keep them along the connectors visually)
> - Some domes will activate a squad's "_on_area_entered" signal, causing the squad to be reassigned to an open slot and not leave the dome. Trying to move the squad again usually works the second time.

File: autoloads/dome_connections.gd [hunk 1]
```diff
@@ -45,6 +45,7 @@ func _add_connection(dome_a: Area2D, dome_b: Area2D):
 		# "distance": 100   # could use to determine squad travel time between domes
 	}
 	connections.append(new_connection)
+	dome_a.connections.append(dome_b)
```
Comment (cmavelis) [inline autoloads/dome_connections.gd:48]: main problem i see with this is you're duplicating information -- keeping all the connection information in this script and querying as need be makes sure it's the source of truth.  This only really becomes important if you still wanted connections to be dynamic, like destroying or building a new connector

I wrote `get_dome_connections` further down this file for what you need in the movement

Comment (avelitj12) [reply autoloads/dome_connections.gd:48]: Very good point. I think I can use that just fine. My bad on missing that. Got lost in the sauce of getting movement working!

Comment (cmavelis) [reply autoloads/dome_connections.gd:48]: no worries, glad you did! key aspect :D

**Review by cmavelis — APPROVED** [review]

**Pull request merged**